front/lib/api/assistant/messages.ts

```diff
@@ -778,9 +778,86 @@ export async function softDeleteUserMessage(
       conversationId: conversation.sId,
       messageId: message.sId,
     },
-    auth.isAdmin()
-      ? "Admin deleted a user message"
-      : "User deleted their message"
+    "User deleted their message"
+  );
+
+  return new Ok({ success: true });
+}
+
+export async function softDeleteAgentMessage(
+  auth: Authenticator,
+  {
+    messageId,
+    conversation,
+  }: {
+    messageId: string;
+    conversation: ConversationWithoutContentType;
+  }
+): Promise<Result<{ success: true }, ConversationError>> {
+  const user = auth.getNonNullableUser();
+  const owner = auth.getNonNullableWorkspace();
+
+  const message = await Message.findOne({
+    where: {
+      conversationId: conversation.id,
+      sId: messageId,
+      workspaceId: owner.id,
+    },
+    include: [
+      {
+        model: AgentMessage,
+        as: "agentMessage",
+        required: true,
+      },
+    ],
+  });
+
+  if (!message || !message.agentMessage) {
+    return new Err(new ConversationError("message_not_found"));
+  }
+
+  if (!message.parentId) {
+    return new Err(new ConversationError("message_deletion_not_authorized"));
+  }
+
+  const parentMessage = await Message.findOne({
+    where: {
+      id: message.parentId,
+      workspaceId: owner.id,
+    },
+    include: [
+      {
+        model: UserMessage,
+        as: "userMessage",
+        required: false,
+      },
+    ],
+  });
```

Comment:
don't we already have a resource that fetches both for us? (just surprised that we don't, and not asking for any refactoring)